### PR TITLE
Improvements to install script.

### PR DESCRIPTION
**Pull request by johndah4x0r**
A ton of improvements to installer script. Uninstall script coming soon.
Let me know if a bug in the script exists.

### CODE REVIEW

File: install_fritzing.sh [hunk 1]
```diff
@@ -50,7 +54,7 @@ for i in $ARGS; do
         '-u' | '--user')
             USERMODE="true"
             ;;
-        '-*' | '--*')
+        -* | --*)
```
Comment (johndah4x0r) [inline install_fritzing.sh]: `case' statements do accept regex, but not inside quotes.

File: install_fritzing.sh [hunk 2]
```diff
@@ -251,18 +260,24 @@ fi
 echo "Doing final touch..."
 
 # Escape all forward slashes to make `sed` stable
-APP_ICON_ESC="$(echo "$APP_ICON" | sed 's/\//\\\//g')"
+APP_ICON_ESC="$(esc_path "$APP_ICON")"
 
 # Modify Fritzing desktop executable [keep original]
-cat fritzing.desktop | sed "s/icons\/fritzing_icon.png/${APP_ICON_ESC}/" > fritzing.use
+sed -i.use "s/icons\/fritzing_icon.png/${APP_ICON_ESC}/" fritzing.desktop
```
Comment (johndah4x0r) [inline install_fritzing.sh]: `shellcheck` is always barking at me that I have the "not-so-orthodox" syntax.

File: install_fritzing.sh [hunk 3]
```diff
@@ -251,18 +260,24 @@ fi
...
 
 # Install Fritzing desktop application
 cp fritzing.use "$APPS"/fritzing.desktop
 cp icons/fritzing_icon.png "$APP_ICON"
 rm -f fritzing.use
 
-#TODO:  Uninstall script.
+# Uninstall script.
```
Comment (johndah4x0r) [inline install_fritzing.sh]: TODO

**Comment by aknoerig** [comment]
Very nice! The in-depth linux experience is appreciated. :)

**Pull request merged**